scipy/optimize/_zeros_py.py

```diff
@@ -1393,3 +1393,189 @@ def toms748(f, a, b, args=(), k=1,
+        is the root and ``r`` is a `RootResults` object.
+        If True and `x0` is non-scalar, the return value is ``(x, converged,
+        zero_der)`` (see Returns section for details).
+    disp : bool, optional
+        If True, raise a RuntimeError if the algorithm didn't converge, with
+        the error message containing the number of iterations and current
+        function value. Otherwise, the convergence status is recorded in a
+        `RootResults` return object.
+        Ignored if `x0` is not scalar.
+        *Note: this has little to do with displaying, however,
+        the `disp` keyword cannot be renamed for backwards compatibility.*
+
+    Returns
+    -------
+    root : float, complex, sequence, or ndarray
+        Estimated location where function is zero.
+    r : `RootResults`, optional
+        Present if ``full_output=True`` and `x0` is scalar.
+        Object containing information about the convergence. In particular,
+        ``r.converged`` is True if the routine converged.
+    converged : ndarray of bool, optional
+        Present if ``full_output=True`` and `x0` is non-scalar.
+        For vector functions, indicates which elements converged successfully.
+    zero_der : ndarray of bool, optional
+        Present if ``full_output=True`` and `x0` is non-scalar.
+        For vector functions, indicates which elements had a zero derivative.
+
+    See Also
+    --------
+    root_scalar : interface to root solvers for scalar functions
+    root : interface to root solvers for multi-input, multi-output functions
+
+    Notes
+    -----
+    The order of convergence of the Muller's method is approximately 1.84.
+    This is slightly faster than secant method, but slower than Newton's
+    method which is quadratic.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from scipy.optimize import muller
+
+    >>> def f(x):
+    ...     return (x**2 + 1)  # roots at x = 1j and x = -1j
+
+    >>> root = muller(f, 0, 1, 2 + 1j)
+    >>> root
+    -1.2168152957711588e-17+1j
+
+    """
+
+    if tol <= 0:
+        raise ValueError(f"tol too small ({tol:g} <= 0)")
+    maxiter = operator.index(maxiter)
+    if maxiter < 1:
+        raise ValueError("maxiter must be greater than 0")
+    if np.size(x0) > 1:
+        raise ValueError("operations on arrays are not yet supported")
+
+    funcalls = 0
+
+    # Ensure inputs are numpy arrays and converted to complex.
+    p0 = np.asarray(x0, dtype=complex)[()] * 1.0
+    p1 = np.asarray(x1, dtype=complex)[()] * 1.0
+    p2 = np.asarray(x2, dtype=complex)[()] * 1.0
+
+    converged = False
+    method = "muller"
+
+    if x0 == x1 or x0 == x2 or x1 == x2:
+        raise ValueError("x0, x1, and x2 must be different")
+
+    f0, f1, f2 = f(p0, *args), f(p1, *args), f(p2, *args)
+    funcalls += 3
+
+    # If one of the guesses is a root, return it immediately.
+    if np.isclose(f0, 0, rtol=rtol, atol=tol):
```

Comment:
Yeah, this is the reason to make a distinction between abscissa and function tolerance. Note that `rtol` has no effect here. In the elementwise functions, `frtol` is relative to the function value at the initial guess.